### PR TITLE
Open redirect on SonarQube in the "login flow"

### PR DESCRIPTION
**Pull request by Mik317**
## ✍️ Description

The `sonarqube` software is vulnerable against `open redirect` on the `login` due to `check bypass` of the `?redirect_to` parameter. The `parameter` is sanitized using some conditional checks that however can be bypassed through a `NULL char encoded`.

## 🕵️‍♂️ Proof of Concept

1. Install the SonarQube software through the website/github repo
2. Start it from the bin directory using ./sonar.sh start
3. Go on http://localhost:9000/sessions/new?return_to=%2F%09%2Fevil.com and login with the default credentials (admin:admin)
4. You'll be redirected on the http://evil.com website

## 💥 Impact

`Open redirect` and probably `referer leaking` issues

## ☎️ Contact

-

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by Mik317** [comment]
Hi @mufeedvh :smile:,
I verified the `open redirect` issue on the `latest version` of the product which can be downloaded by the website, so I'm sure it exists (I'll provide a video PoC if needed).

Regarding the `hyperlink`, I updated that one which I found through `search` directly with some I think should point to the dangerous code (are in the `master` branch).

Cheers,
Mik

**Comment by mufeedvh** [comment]
Thanks for the clarification and the update @Mik317! :+1:

**Comment by mufeedvh** [comment]
Thanks for the clarification and update @Mik317! :+1:

File: bounties/maven/sonarqube/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "PackageVulnerabilityID": "1",
+  "DisclosureDate": "2020-08-04",
+  "AffectedVersionRange": "*",
+  "Summary": "Open redirect in the \"login flow\" due to \"?redirect_to\" parameter check bypass",
+  "Author": {
```
[the remark this answers is not in the document]
Comment (Mik317) [reply bounties/maven/sonarqube/1/vulnerability.json]: Done :cake: 

Cheers,
Mik

**Pull request merged**